### PR TITLE
Add highest value method for computing probability of winning

### PR DESCRIPTION
**Pull request by robertjneal**
Also included in this PR are the changes necessary for pushing to Sonatype / Maven Central

### CODE REVIEW

File: src/test/scala/com/udemy/statistics/algorithm/MonteCarloSpec.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  private val sampleD = "sample D"
+
+  private val datasetA = sampleA -> NonEmptyList(100, 200, 30, 44, 51, 160, 740, 10, 430, 3)
+  private val datasets = Map(
+    datasetA,
+    sampleB -> NonEmptyList(110, 210, 40, 54, -61, 170, 750, 20, 440, 13),
+    sampleC -> NonEmptyList(98, 150, 56, -23, 98, 120, 660, -135, 430, -12)
+  )
+
+  "dataset winProbability" should "appropriately distribute wins to better datasets" in {
```
Comment (pauljcho) [inline src/test/scala/com/udemy/statistics/algorithm/MonteCarloSpec.scala]: the method name isn't `winProbability` anymore

File: src/main/scala/com/udemy/statistics/algorithm/MonteCarlo.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import com.udemy.statistics.distribution.Normal
+import com.udemy.statistics.Probability
+import org.apache.commons.math3.distribution._
+import scalaz.NonEmptyList
+
+import scala.annotation.tailrec
+
+object MonteCarlo {
+
+  def highestValue[T:Numeric](options: Map[String, NonEmptyList[T]], draws: Int): Map[String, Probability] =
```
[the remark this answers is not in the document]
Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/MonteCarlo.scala]: I take it that the function is returning a collection of probabilities of a particular option having the highest value, where value means something like most likely to have the highest value if randomly sampled.

You could say that `win` means `highest value`, but I don't think the function is necessarily saying something about one thing winning or not winning. My thinking is I have a function that I pass a set of options and it returns me the probability that each of those has the highest value. I do think `highest value` might be too encompassing or too vague. On the other hand I think `win` is too specific.

I'm open to suggestions, but I think the function name needs to say what the probabilities being returned represent.

[the remark this answers is not in the document]
Comment (pauljcho) [reply src/main/scala/com/udemy/statistics/algorithm/MonteCarlo.scala]: I see. is it redundant to say something like highestValueProbability?

[the remark this answers is not in the document]
Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/MonteCarlo.scala]: I use `probablityOfBeingBest` as a private def in this file. What do you think about that name? I could make the public defs match if you think it's appropriately descriptive.

[the remark this answers is not in the document]
Comment (pauljcho) [reply src/main/scala/com/udemy/statistics/algorithm/MonteCarlo.scala]: Yeah that sounds good to me

File: src/test/scala/com/udemy/statistics/algorithm/MonteCarloSpec.scala [hunk 3]
```diff
@@ -37,31 +37,31 @@ class MonteCarloSpec extends FlatSpec with Matchers {
     sampleC -> NonEmptyList(98, 150, 56, -23, 98, 120, 660, -135, 430, -12)
   )
 
-  "dataset winProbability" should "appropriately distribute wins to better datasets" in {
+  "datasets" should "appropriately distribute wins to better datasets" in {
```
Comment (pauljcho) [inline src/test/scala/com/udemy/statistics/algorithm/MonteCarloSpec.scala]: i think subjects of `should` for these tests should be the name of the function you're testing

**Review by pauljcho — APPROVED** [review]

**Pull request merged**